src/BuildPrediction/Predictors/AssemblyOriginatorKeyFilePredictor.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Build.Prediction.Predictors
+{
+    using System;
+    using Microsoft.Build.Evaluation;
+    using Microsoft.Build.Execution;
+
+    /// <summary>
+    /// Predicts inputs for the key file used for strong naming.
+    /// </summary>
+    public class AssemblyOriginatorKeyFilePredictor : IProjectPredictor
+    {
+        internal const string SignAssemblyPropertyName = "SignAssembly";
+
+        internal const string AssemblyOriginatorKeyFilePropertyName = "AssemblyOriginatorKeyFile";
+
+        /// <inheritdoc/>
+        public void PredictInputsAndOutputs(
+            Project project,
+            ProjectInstance projectInstance,
+            ProjectPredictionReporter predictionReporter)
+        {
+            // The key file is only an input when signing is enabled.
+            var signAssembly = projectInstance.GetPropertyValue(SignAssemblyPropertyName);
+            if (!signAssembly.Equals("true", StringComparison.OrdinalIgnoreCase))
+            {
+                return;
+            }
+
+            var assemblyOriginatorKeyFile = projectInstance.GetPropertyValue(AssemblyOriginatorKeyFilePropertyName);
+            if (!string.IsNullOrEmpty(assemblyOriginatorKeyFile))
+            {
+                predictionReporter.ReportInputFile(assemblyOriginatorKeyFile);
+            }
```

Comment:
Allowed to not exist? If not, need a File.Exists check

Reply:
Today QuickBuild does check for existence commenting that the C# compiler is tolerant of the file not existing, but when I actually tried it I found that to not be the case:

```
CSC : error CS7027: Error signing output with public key from file 'E:\MSBuildPrediction\\build\StrongNaming.snk' -- Could not find file 'E:\MSBuildPrediction\build\StrongNaming.snk'. [E:\MSBuildPrediction\src\BuildPrediction\Microsoft.Build.Prediction.csproj]
```

Based on that I decided to avoid the existence check.